### PR TITLE
Update up.ps1

### PR DESCRIPTION
**Pull request by mykolasredin**
Added Login flow without UI, added parameters for required variables.

### CODE REVIEW

File: up.ps1 [hunk 1]
```diff
@@ -1,15 +1,23 @@
+Param (
+    [switch]$SilentLogin,
```
Comment (AndreyPilipchenko) [inline up.ps1]: This variables or placeholders for them should be defined in .env file, and assigned in scrip body, not in params

File: up.ps1 [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
+    [string]$xmCloudAud
+)
 $ErrorActionPreference = "Stop";
 
 $envContent = Get-Content .env -Encoding UTF8
 $xmCloudHost = $envContent | Where-Object { $_ -imatch "^CM_HOST=.+" }
 $xmCloudDeployConfig = $envContent | Where-Object { $_ -imatch "^XMCLOUDDEPLOY_CONFIG=.+" }
 $sitecoreDockerRegistry = $envContent | Where-Object { $_ -imatch "^SITECORE_DOCKER_REGISTRY=.+" }
 $sitecoreVersion = $envContent | Where-Object { $_ -imatch "^SITECORE_VERSION=.+" }
+$xmCloudDomain = $envContent | Where-Object { $_ -imatch "^SITECORE_FedAuth_dot_Auth0_dot_Domain=.+" }
```
Comment (AndreyPilipchenko) [inline up.ps1]: All variables, related to client login should be defined here 

if($SilentLogin){
#define variables here
}

File: up.ps1 [hunk 3]
```diff
@@ -66,7 +74,12 @@ foreach ($pluginJsonFile in $pluginJsonFiles) {
 }
 
 Write-Host "Logging into Sitecore..." -ForegroundColor Green
-dotnet sitecore cloud login
+if ($SilentLogin) {
+    dotnet sitecore cloud login --authority $xmCloudDomain --audience $xmCloudAud --client-id $AuthClientId --client-secret $AuthClientSecret --client-credentials
```
Comment (AndreyPilipchenko) [inline up.ps1]: add --allow-write true

File: up.ps1 [hunk 4]
```diff
@@ -66,7 +74,12 @@ foreach ($pluginJsonFile in $pluginJsonFiles) {
...
+}
+else {
+    dotnet sitecore cloud login
+}
 dotnet sitecore login --ref xmcloud --cm https://$xmCloudHost --allow-write true
```
Comment (AndreyPilipchenko) [inline up.ps1]: move under else{}

File: .env [hunk 5]
```diff
@@ -63,7 +63,9 @@ MEDIA_REQUEST_PROTECTION_SHARED_SECRET=
 # SITECORE_FedAuth_dot_Auth0_dot_ClientSecret=nNcrp1x4kEkkxu-LhChEcBvlosdQ0_Y7gfnUfKwkBitM9MR2SBc4UPCRP19SGf8D
 # SITECORE_FedAuth_dot_Auth0_dot_Domain=https://auth-staging-1.sitecore-staging.cloud
 # SITECORE_FedAuth_dot_Auth0_dot_Audience=https://api-staging.sitecore-staging.cloud;https://xmcloud-cm-staging.sitecore-staging.cloud
-# SITECORE_FedAuth_dot_Auth0_dot_LogoutRedirect=/sitecore
+# SITECORE_FedAuth_dot_Auth0_dot_LogoutRedirect=/sitecore
+# SITECORE_FedAuth_dot_Auth0_dot_SilentLogin=false
```
Comment (AndreyPilipchenko) [inline .env]: Please change name to_ClientCredentialsLogin and also change it in script

File: .env [hunk 6]
```diff
@@ -73,6 +75,8 @@ SITECORE_FedAuth_dot_Auth0_dot_ClientSecret=qasWWubfPROthbtZoaK4mV_TKpb3Ya50oBQX
 SITECORE_FedAuth_dot_Auth0_dot_Domain=https://auth-beta.sitecorecloud.io
 SITECORE_FedAuth_dot_Auth0_dot_Audience=https://api-beta.sitecorecloud.io;https://xmcloud-cm-beta.sitecorecloud.io
 SITECORE_FedAuth_dot_Auth0_dot_LogoutRedirect=/sitecore
+SITECORE_FedAuth_dot_Auth0_dot_SilentLogin=false
+SITECORE_FedAuth_dot_Auth0_dot_AudienceForSilentLogin=
```
Comment (AndreyPilipchenko) [inline .env]: is it really needed? an audience should be the same for all login variants

File: README.md [hunk 7]
```diff
@@ -124,6 +124,7 @@ See Sitecore Containers documentation for more information on system requirement
 
 1. When prompted, log into Sitecore via your browser, and
    accept the device authorization.
+    * To log in via client credentials flow, set the environment variable SITECORE_FedAuth_dot_Auth0_dot_SilentLogin to "true" and update values of the correspond environment variables (Domain, AudienceForSilentLogin, ClientId, ClientSecret)
```
Comment (AndreyPilipchenko) [inline README.md]: Please provide exact names of variables to update

File: up.ps1 [hunk 8]
```diff
@@ -66,8 +79,13 @@ foreach ($pluginJsonFile in $pluginJsonFiles) {
 }
 
 Write-Host "Logging into Sitecore..." -ForegroundColor Green
-dotnet sitecore cloud login
-dotnet sitecore login --ref xmcloud --cm https://$xmCloudHost --allow-write true
+if ($SilentLogin -eq "true") {
+    dotnet sitecore cloud login --authority $xmCloudDomain --audience $xmCloudAudSilentLogin --client-id $xmCloudClientId --client-secret $xmCloudClientSecret --client-credentials
```
Comment (AndreyPilipchenko) [inline up.ps1]: please add --allow-write true

File: up.ps1 [hunk 9]
```diff
@@ -66,8 +79,13 @@ foreach ($pluginJsonFile in $pluginJsonFiles) {
...
+}
+else {
+    dotnet sitecore cloud login
+	dotnet sitecore login --ref xmcloud --cm https://$xmCloudHost --allow-write true
```
Comment (AndreyPilipchenko) [inline up.ps1]: fix idents

**Review by AndreyPilipchenko — CHANGES_REQUESTED** [review]

File: .env (hunk 6 above)
Comment (mykolasredin) [reply .env]: We have all variables doubled in this script, e.g., SITECORE_FedAuth_dot_Auth0_dot_IsLocal

File: up.ps1 (hunk 8 above)
Comment (mykolasredin) [reply up.ps1]: The script failed, if I add this parameter.

**Pull request merged**